# misEval #
Misassembly Evaluator

----------
misEval is a tool that aims to evaluate the misassembly intervals.it receives the signals from the given intervals to judge whether the intervals include assembly errors.

## Prerequisites ##
misEval depends on the following libraries and tools:
1. [HTSlib(V1.9 or later)](https://github.com/samtools/htslib)
2. [mlpack(v3.4.2 or later)](https://github.com/mlpack/mlpack)
3. g++ (v4.7 or later which supports c++11).
4. python3
5. autotools

## Quick install ##
* Debian / Ubuntu 
```
$ sudo apt update  # Ensure the package list is up to date
$ sudo apt install mlpack-bin g++ python3 autoconf automake libtool
```
HTSlib needs to be installed from source files.

* RedHat / CentOS
```
$ sudo yum install g++ python3 autoconf automake libtool
```
HTSlib and mplack need to be installed from source files.

## Compiling misEval ##

The binary file can be generated by typing:
```sh
$ git clone https://github.com/zhuxiao/miseval.git
$ sudo cp miseval.py /usr/local/bin
$ cd misEval/src/
$ libtoolize
$ aclocal
$ autoheader
$ automake --add-missing
$ automake
$ autoconf
$ ./configure
$ make
$ sudo make install
```
And the binary file `miseval` will be output into the folder `usr/local/bin` in this package directory.

## misEval command ##

Simplyï¼ŒmisEval can be run by the command:

$ miseval.py -r referenceSequence.fa -b bamFile.bam -f regionFile

Then, whether the intervals include assembly errors will be marked:

Program: miseval v0.3.0

    Usage:  miseval.py [options] -r <REF> -b <BAM> -r <REG>

    Description:
        REF     Reference file (required)
        BAM     Coordinate-sorted BAM file (required)
        REG     Regions to be analyzed: CHR|CHR:START-END.(required)
    Inputs:
        -r PATH        reference file
        -b PATH        bam file
        -f PATH        regions file
    General:
        -min-regsize INT    [200]
            the evaluated region will be extended to INT bp symmetrically 
            when it is shorter than INT bp. 
        -extend-regsize-fold    FLOAT   [2]
            auxiliary region is used to set baseline for an evaluated region, 
            it is determined by extending the evaluated region by 
            FLOAT*(the evaluated region size) on both sides. 

    Mode: 
        -cov    on/off    coverage clustering   [on]
            --min-cov-fold  FLOAT   [0.5]
                the coverage is anomalous when it is lower than FLOAT*meancov,
                 where meancov is the average coverage of auxiliary region.
            --max-cov-fold  FOLAT   [2]
                the coverage is anomalous when it is higher than FLOAT*meancov.
        -indels on/off   indels clustering      [on]
            --min-locs-ratio    FLOAT   [0.2]
                mask the ultra-low coverage locations whose coverage is lower 
                than FLOAT*meancov. when indels appear at ultra-low coverage,
                the performance on indels is unreliable.
        -abstrand   on/off   anomalous orientation reads clustering     [0.2]
            --min-abstrand-ratio    FLOAT   [0.3]
                search the anomalous orientation clumps in which the anomalous 
                orientation reads take a higher ratio than FLOAT at each location.
        -abisize    on/off    anomalous insertsize reads clustering     [on]
            --min-abisize-ratio     FLOAT   [0.3]
                search the anomalous insertsize clumps in which the anomalous 
                insertsize reads take a higher ratio than FLOAT at each location.
            --isize-sdev-fold FLOAT     [3]
                the anomalous insertsize is short than (the average insertsize)-FLOAT*sdev bp 
                or longer than  (the average insertsize)+FLOAT*sdev bp, otherwise it is anomalous.
        -abmate     on/off   chromosomal rearrangement reads clustering     [on]
            --min-abmate-ratio  FLOAT   [0.3]
                search the chromosomal rearrangement clumps in which the chromosomal 
                rearrangement reads take a higher ratio than FLOAT at each location.
   
    Options:
        -o PATH             outdir
        -h                  show this help message and exit


## Contact ##

If you have problems or some suggestions, please contact: [email](zhuxiao_hit@yeah.net) without hesitation. 

---- Enjoy !!! -----

